addons/l10n_in_ewaybill_stock/models/stock_picking.py

```diff
@@ -14,16 +13,6 @@ def _get_l10n_in_ewaybill_form_action(self):
 
     def action_l10n_in_ewaybill_create(self):
         self.ensure_one()
-        if (
-            product_with_no_hsn := self.move_ids.mapped('product_id').filtered(
-                lambda product: not product.l10n_in_hsn_code
-            )
-        ):
-            raise UserError(_("Please set HSN code in below products: \n%s", '\n'.join(
-                [product.name for product in product_with_no_hsn]
-            )))
-        if self.l10n_in_ewaybill_id:
-            raise UserError(_("Ewaybill already created for this picking."))
```

Comment:
Can we have better way? what about a constraint?

Reply:
Hello @hamo-odoo ,
I don't think there is any need for this validation. This action is triggered from the 'Create E-waybill' button and that button will be hidden if `l10n_in_ewaybill_id` is available also there is no other way to create an e-waybill again.